File: backend/app/schemas/matching.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class TmdbCandidate(BaseModel):
    tmdb_id: int
    title: str
    original_title: str | None
    first_air_date: str | None
    overview: str
    poster_path: str | None
    match_score: float


class TmdbSearchResponse(BaseModel):
    message: str
    candidates: list[TmdbCandidate]


class MediaMatchSaveRequest(BaseModel):
    tmdb_id: int | None = None
    title: str | None = None
    original_title: str | None = None
    year: int | None = Field(default=None, ge=0)
    season_number: int | None = Field(default=None, ge=0)
    episode_number: int | None = Field(default=None, ge=0)
    episode_title: str | None = None
    match_score: float = Field(default=0, ge=0, le=100)
    status: str = "pending"


class MediaMatchResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    source_item_id: int
    tmdb_id: int | None
    title: str | None
    original_title: str | None
    year: int | None
    season_number: int | None
    episode_number: int | None
    episode_title: str | None
    match_score: float
    status: str
    created_at: datetime
    updated_at: datetime


class MediaMatchMessageResponse(BaseModel):
    message: str
    match: MediaMatchResponse


class MediaMatchListResponse(BaseModel):
    message: str
    matches: list[MediaMatchResponse]

File: frontend/src/types/matching.ts
export interface TmdbCandidate {
  tmdb_id: number
  title: string
  original_title: string | null
  first_air_date: string | null
  overview: string
  poster_path: string | null
  match_score: number
  search_query: string
}

export interface TmdbSearchResponse {
  message: string
  search_queries: string[]
  candidates: TmdbCandidate[]
}

export interface MediaMatchPayload {
  tmdb_id: number | null
  title: string | null
  original_title: string | null
  year: number | null
  season_number: number | null
  episode_number: number | null
  episode_title: string | null
  match_score: number
  status: string
}

export interface MediaMatch {
  id: number
  source_item_id: number
  tmdb_id: number | null
  title: string | null
  original_title: string | null
  year: number | null
  season_number: number | null
  episode_number: number | null
  episode_title: string | null
  match_score: number
  status: string
  created_at: string
  updated_at: string
}

export interface MediaMatchMessageResponse {
  message: string
  match: MediaMatch
}

export interface MediaMatchListResponse {
  message: string
  matches: MediaMatch[]
}
